.travis.yml

```diff
@@ -1,28 +1,65 @@
 language: scala
+
 scala:
    - 2.11.12
+
 jdk:
  - oraclejdk8
 
-# These directories are cached to S3 at the end of the build
-cache:
-  directories:
-    - $HOME/.ivy2/cache
-    - $HOME/.sbt/boot/
+sudo: required
+
+services:
+  - docker
+
+addons:
+  chrome: stable
+
+env:
+  global:
+    - APPLICATION_NAME=nussknacker
+    - SANITIZED_BRANCH=`echo ${TRAVIS_BRANCH} | sed 's/[^a-zA-Z0-9]/\_/g' | awk '{print tolower($0)}'`
+    - BUILD_VERSION=$(if [[ -n $TRAVIS_TAG ]]; then echo "${SANITIZED_BRANCH}"; else echo "${SANITIZED_BRANCH}-$TRAVIS_COMMIT"; fi)
+
+before_install:
+  - if [[ "$TRAVIS_PULL_REQUEST" == "false" ]]; then echo "$DOCKER_PASSWORD" | docker login -u "$DOCKER_USERNAME" --password-stdin; fi
 
 before_cache:
   # Cleanup the cached directories to avoid unnecessary cache updates
   - rm -fv $HOME/.ivy2/.sbt.ivy.lock
   - find $HOME/.ivy2/cache -name "ivydata-*.properties" -print -delete
   - find $HOME/.sbt        -name "*.lock"               -print -delete
 
-script:
- - sbt ++$TRAVIS_SCALA_VERSION clean test management/it:test
-
-sudo: required
+# These directories are cached to S3 at the end of the build
+cache:
+  directories:
+    - $HOME/.ivy2/cache
+    - $HOME/.sbt/boot/
 
-services:
-  - docker
+stages:
+  - tests
+  - name: releases
+    if: fork = false AND type != cron
+  - name: deploys
+    if: branch IN (demo) AND fork = false AND type != cron
 
-addons:
-  chrome: stable
\ No newline at end of file
+jobs:
+  include:
+    - stage: tests
+      name: "Unit tests"
+      script: sbt ++$TRAVIS_SCALA_VERSION clean test
+    - name: "Management tests"
+      script: sbt ++$TRAVIS_SCALA_VERSION clean management/it:test
+    - name: "EngineStandalone tests"
+      script: sbt ++$TRAVIS_SCALA_VERSION clean engineStandalone/it:test
+    - name: "UI slow tests"
+      script: sbt ++$TRAVIS_SCALA_VERSION clean assemblySamples ui/slow:test
+    - name: "Tests docker demo"
+      script: cd ./demo/docker && ./testQuickstart.sh
+    - stage: releases
```

Comment:
Shouldn't it be made parallel with tests? It has no dependency to previous stage. Maybe first stage should be "build" which will do parallel tests and real build.